gpytorch/priors/smoothed_box_prior.py

```diff
@@ -71,3 +71,20 @@ def _log_prob(self, x):
         # x = "distances from box`"
         X = ((x - self._c).abs_() - self._r).clamp(min=0)
         return (self.tails.log_prob(X) - self._M).sum(-1)
+
+    def sample(self, sample_shape=torch.Size()):
```

Comment:
I think this could be a `rsample` method - `a` and `b` can be differentiated through given your implementation. (Not that I see the use case right now, but it could be a nice future-proof addition).

Reply:
Indeed, I got hung up on the `torch.bernoulli` for some reason and couldn't see that the method could've been `rsample` from the start!